src/compiler/crystal/tools/init.cr

```diff
@@ -56,6 +56,25 @@ DIR  - directory where project will be generated,
       github_user.empty? ? default : github_user
     end
 
+    def self.fetch_name(opts, args)
+      name = fetch_required_parameter(opts, args, "NAME")
+      name_type = fetch_name_type(name)
+      if name_type
+        puts "invalid NAME value: #{name} #{name_type} already exists"
```

Comment:
"foo directory/file already exists", doesn't read that backwards? Also what about printing to stderr?